File: JavaScript/Codility/NumberOfDiscIntersections.js
function solution(A) {
    var newArray = [];
    for (var i=0; i < A.length; i++){
        newArray.push({a:i-A[i], b:i+A[i]});
    }
    newArray.sort((a,b) => (a.a-b.a));

    var result = 0;
    for (var i=0; i < newArray.length-1; i++){
        for (var j=i+1; j <newArray.length; j++){
            if (newArray[i].b >= newArray[j].a){
                result++;
            }
            else{
                break;
            }
            if (result > 10000000){
                return -1;
            }
        }
    }
    return result;
}

File: Python/Codility/NumberOfDiscIntersections.py
def solution(A):
    disc = []
    for idx, val in enumerate(A):
        disc.append((idx - val, -1))
        disc.append((idx + val, 1))
    disc.sort()

    intersect = 0
    intervals = 0
    for d in disc:
        if d[1] == 1:
            intervals -= 1
        else:
            intersect += intervals
            intervals += 1

    return -1 if intersect > 10000000 else intersect